chore(expo): Remove all react canary support now that React 19 is the default

EvanBacon opened the pull request:
# Why

- We no longer need canary builds of React to develop React Server Components or owner stack. We may need to add this back in the future when some new React feature comes out, but for now it's just adding a layer of untested complexity that no one is actively maintaining and currently adds nothing.
- The flag will now warn that nothing is happening and that you can remove the flag.

# Test Plan

- Ran the three RSC tests on iOS, Android, and web. All appeared fine. We were close last round but the sibling issue in React suspense was breaking native RSC.

EvanBacon committed:
Remove all react canary support now that React 19 is the default

EvanBacon committed:
changelog

kitten reviewed (approved):
🎉💯 https://maps.app.goo.gl/FzYDHcfXEF9xnFXL7

kitten committed:
test(cli): Fix bun isolated monorepo E2E tests and `moduleMapper.test.ts` (#40394)

Bun switched to 1.3, which defaults to isolated installations which then
fails our monorepo install tests. Basing a removal of the affected
`getConfig` onto #40386 should fix the remaining E2E issue.

EvanBacon committed:
Merge branch 'main' into @evanbacon/cli/remove-react-canary
